tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -903,8 +901,7 @@ impl LeaderTpuService {
 
             if cache_update_info.has_some() {
                 let mut leader_tpu_cache = leader_tpu_cache.write().unwrap();
-                let (has_error, cluster_refreshed) = leader_tpu_cache
-                    .update_all(recent_slots.estimated_current_slot(), cache_update_info);
```

Comment:
we no longer recompute estimated current slot here. Instead, we use the value that is now stored in `cache_update_info` which is in sync w/ the rest of the update info